### PR TITLE
tests: arithmetic field operation integration tests

### LINKED ISSUE #165: Integration tests for arithmetic field operations
Tests for arithmetic field operations need to be added to the [integration tests](https://github.com/maticnetwork/miden/blob/next/miden/tests/integration/operations/field_ops.rs).

Tests should follow the existing pattern with the following tests for each of the arithmetic field operations that are missing tests:
- a test of simple success cases
- a test with applicable failure cases, if they exist (compilation and/or execution failure)
- a proptest of the expected success case

The [u32 arithmetic tests](https://github.com/maticnetwork/miden/blob/next/miden/tests/integration/operations/u32_ops/arithmetic_ops.rs) provide a fairly complete example.

Specs for the operations are in the [Miden Assembly note](https://hackmd.io/YDbjUVHTRn64F4LPelC-NA).

The field operations which are missing tests are:

- [x] add
- [x] add.b
- [x] sub
- [x] sub.b
- [x] mul
- [x] mul.b
- [x] div
- [x] div.b
- [x] neg
- [x] inv

More information can also be seen in the [field_ops assembly parsers](https://github.com/maticnetwork/miden/blob/next/assembly/src/parsers/field_ops.rs).

### PR DESCRIPTION
**Pull request by timgestson**
Added arithmetic field operation integration tests to fix #165

### CODE REVIEW

File: miden/tests/integration/operations/field_ops.rs [hunk 1]
```diff
@@ -1,12 +1,255 @@
...
+    let test = build_op_test!(asm_op, &[1, 0]);
+    test.expect_stack(&[0]);
+
+    let test = build_op_test!(asm_op, &[1, 5]);
+    test.expect_stack(&[5]);
+
+    // --- test overflow --------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[Felt::MODULUS, 2]);
+    let expected = Felt::MODULUS as u128 * 2 as u128 % Felt::MODULUS as u128;
+    test.expect_stack(&[expected as u64]);
```
Comment (bobbinth) [inline miden/tests/integration/operations/field_ops.rs:109]: `Felt::MODULUS` gets reduced to `0` - so, this is basically testing multiplication by `0`. Maybe use `Felt::MODULUS - 1` instead?

File: miden/tests/integration/operations/field_ops.rs [hunk 2]
```diff
@@ -1,12 +1,255 @@
...
+    let test = build_op_test!(build_asm_op(0), &[1]);
+    test.expect_stack(&[0]);
+
+    let test = build_op_test!(build_asm_op(1), &[5]);
+    test.expect_stack(&[5]);
+
+    // --- test overflow --------------------------------------------------------------------------
+    let test = build_op_test!(build_asm_op(2), &[Felt::MODULUS]);
+    let expected = Felt::MODULUS as u128 * 2 as u128 % Felt::MODULUS as u128;
+    test.expect_stack(&[expected as u64]);
```
Comment (bobbinth) [inline miden/tests/integration/operations/field_ops.rs:131]: Similar comment to above re using `Felt::MODULUS - 1`.

Comment (bobbinth) [inline miden/tests/integration/operations/field_ops.rs:131]: Similar comment to above re using `Felt::MODULUS - 1`.

File: miden/tests/integration/operations/field_ops.rs [hunk 3]
```diff
@@ -1,12 +1,255 @@
...
+    // --- simple cases ---------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[0, 1]);
+    test.expect_stack(&[0]);
+
+    let test = build_op_test!(asm_op, &[2, 1]);
+    test.expect_stack(&[2]);
+
+    // --- test remainder -------------------------------------------------------------------------
+    let test = build_op_test!(asm_op, &[5, 2]);
+    let expected = (Felt::new(2).inv().as_int() as u128 * 5 as u128) % Felt::MODULUS as u128;
```
Comment (bobbinth) [inline miden/tests/integration/operations/field_ops.rs:152]: Very minor nit: instead of `5 as u128` you could probably say `5_u128`.

Comment (bobbinth) [inline miden/tests/integration/operations/field_ops.rs:152]: Very minor nit: instead of `5 as u128` you could probably say `5_u128`.

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks great! thank you so much! There are a couple of very minor issues (I left comments inline).

File: miden/tests/integration/operations/field_ops.rs (hunk 1 above)
Comment (timgestson) [reply miden/tests/integration/operations/field_ops.rs]: Oh right good catch.  Fixed

**Review by bobbinth — APPROVED** [review]
Looks great! Thank you again!

**Pull request merged**